clusterloader2/pkg/util/measurement_executor.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"strings"
+)
+
+// ExecuteMeasurement executes a measurement, which can be a single measurement or a wrapper for multiple measurements.
+// It throws an error if the Identifier type is neither string nor a map of string->params
+func ExecuteMeasurement(mm measurement.MeasurementManager, methodName string, identifier interface{}, params map[string]interface{}) error {
+	switch identifier.(type) {
+	case string:
+		return executeSingleMeasurement(mm, methodName, identifier.(string), params)
+	case map[string]interface{}:
+		return executeWrapperMeasurement(mm, methodName, identifier.(map[string]interface{}), params)
+	default:
+		return fmt.Errorf("invalid type for Identifier. Valid types are string and map[string]interface{}")
+	}
+}
+
+func executeSingleMeasurement(mm measurement.MeasurementManager, methodName string, identifier string, params map[string]interface{}) error {
+	err := mm.Execute(methodName, identifier, params)
+	if err != nil {
+		klog.Warning("Error while executing Measurement method - %s, identifier - %s, err - %v", methodName, identifier, err)
+	}
+	return err
+}
+
+func executeWrapperMeasurement(mm measurement.MeasurementManager, methodName string, identifiers map[string]interface{}, params map[string]interface{}) error {
+
+	var wg wait.Group
+	errList := errors.NewErrorList()
+
+	for identifier := range identifiers {
+
+		// We will add the identifier specific configs, if there are defined, to the params map.
+		if identifiers[identifier] != nil {
+			for configKey, configVal := range identifiers[identifier].(map[string]interface{}) {
+				params[configKey] = strings.TrimSpace(configVal.(string))
```

Comment:
Shouldn't we make a copy of params for each iteration? Otherwise if you have

```
Identifier:
  A:
    Param1: X
  B:
    Param2: Y
```

The measurement B will be called with Params {Param1: X, Param2: Y), right?

Reply:
yeah again an oversight by me :( made the changes.